siliconcompiler/report/streamlit_report.py

```diff
@@ -0,0 +1,113 @@
+import os
+import time
+import tempfile
+import json
+
+try:
+    from streamlit.web import bootstrap
+    from streamlit import config as _config
+    _has_streamlit = True
+except ModuleNotFoundError:
+    _has_streamlit = False
```

Comment:
This seems counterproductive to what you are trying to achieve.
If I try to attach `chip._dashboard()` to the `heartbeat.py` example this leads to no `ModuleNotFoundError` being thrown.
Instead I'd suggest to just rely on Python's built in way of handling modules.
```suggestion
from streamlit.web import bootstrap
from streamlit import config as _config
```
That way I actually got this error as you originally intended I assume:
`ModuleNotFoundError: No module named 'streamlit'`